langcodes/__init__.py

```diff
@@ -741,49 +727,59 @@ def find_name(tagtype: str, name: str, language: {str, 'LanguageData'}):
         LookupError. If more than one subtag is found with the same name, you get
         an AmbiguousError.
 
-        The `language` parameter is the language code or LanguageData object
+        The `language` parameter is the language code or Language object
         representing the language that you're providing the name in.
 
-        >>> LanguageData.find_name('language', 'francés', 'es')
-        LanguageData(language='fr')
+        >>> Language.find_name('language', 'francés', 'es')
+        Language.make(language='fr')
 
-        >>> LanguageData.find_name('region', 'United Kingdom', LanguageData.get('en'))
-        LanguageData(region='GB')
+        >>> Language.find_name('region', 'United Kingdom', Language.get('en'))
+        Language.make(region='GB')
 
-        >>> LanguageData.find_name('script', 'Arabic', 'en')
-        LanguageData(script='Arab')
+        >>> Language.find_name('script', 'Arabic', 'en')
+        Language.make(script='Arab')
 
-        >>> LanguageData.find_name('language', 'norsk bokmål', 'no')
-        LanguageData(language='nb')
+        >>> Language.find_name('language', 'norsk bokmål', 'no')
+        Language.make(language='nb')
 
-        >>> LanguageData.find_name('language', 'norsk bokmal', 'no')
+        >>> Language.find_name('language', 'norsk bokmal', 'no')
         Traceback (most recent call last):
             ...
         LookupError: Can't find any language named 'norsk bokmal'
+
+        Some langauge names resolve to more than a language. For example,
+        the name 'Brazilian Portuguese' resolves to a language and a region,
+        and 'Simplified Chinese' resolves to a language and a script. In these
+        cases, a Language object with multiple subtags will be returned.
+
+        >>> Language.find_name('language', 'Brazilian Portuguese', 'en')
+        Language.make(language='pt', region='BR')
+        >>> Language.find_name('language', 'Simplified Chinese', 'en')
+        Language.make(language='zh', script='Hans')
         """
-        und = LanguageData()
-        english = LanguageData(language='en')
+        und = Language.make()
+        english = Language.make(language='en')
 
         options = DB.lookup_name_in_any_language(tagtype, name)
-        if isinstance(language, LanguageData):
+        if isinstance(language, Language):
             target_language = language
         else:
-            target_language = LanguageData.get(language)
+            target_language = Language.get(language)
         best_options = []
-        best_match_score = 1
+        best_score = 40
```

Comment:
This effectively raises the minimum match score when looking up language names, now that many language pairs match with strength 16 instead of 0. For example, English and French match with strength 16, and that doesn't mean that we should always try French names when looking up English names.